src/components/media-player/ha-browse-media-tts.ts

```diff
@@ -143,31 +102,56 @@ class BrowseMediaTTS extends LitElement {
         );
         const message = params.get("message");
         const language = params.get("language");
-        const gender = params.get("gender");
+        const voice = params.get("voice");
         if (message) {
           this._message = message;
         }
-        if (language && gender) {
-          this._cloudOptions = [language, gender];
+        if (language) {
+          this._language = language;
         }
-      }
-
-      if (this.isCloudItem && !this._cloudTTSInfo) {
-        getCloudTTSInfo(this.hass).then((info) => {
-          this._cloudTTSInfo = info;
-        });
-        fetchCloudStatus(this.hass).then((status) => {
-          if (status.logged_in) {
-            this._cloudDefaultOptions = status.prefs.tts_default_voice;
-            if (!this._cloudOptions) {
-              this._cloudOptions = { ...this._cloudDefaultOptions };
+        if (voice) {
+          this._voice = voice;
+        }
+        const provider = getProviderFromTTSMediaSource(
+          this.item.media_content_id
+        );
+        if (provider !== this._provider?.engine_id) {
+          this._provider = undefined;
+          getTTSEngine(this.hass, provider).then((engine) => {
+            this._provider = engine.provider;
+            if (
+              !this._language &&
+              engine.provider.supported_languages?.length
+            ) {
+              const langRegionCode =
```

Comment:
Maybe use `Intl.DisplayNames.supportedLocalesOf`

Reply:
Can't we pass the country to the endpoint like this list endpoint ?

Reply:
We can... but we don't want to filter it like with the other endpoint, only want to sort...